File: app/src/main/java/uk/co/jatra/lifecycledialog/MainActivity.java
package uk.co.jatra.lifecycledialog;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;
import android.widget.TextView;

import uk.co.jatra.lifecycledialog.LiveDialogFragment.LiveDialog;

public class MainActivity extends AppCompatActivity {

    public static final String TEXT1 = "TEXT1";
    private static final String KEY = "KEY1";
    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        Button button = findViewById(R.id.button);

        if (savedInstanceState != null) {
            textView.setText(savedInstanceState.getString(TEXT1));
        }

        final LiveDialog liveDialog = new LiveDialog(this, KEY)
                .title(R.string.dialog_title)
                .message(R.string.dialog_message)
                .negative(R.string.dialog_negative_button_label)
                .positive(R.string.dialog_positive_button_label)
                .observe(answer -> textView.setText(answer.name()));

        button.setOnClickListener(view -> liveDialog.display());
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        outState.putString(TEXT1, textView.getText().toString());
        super.onSaveInstanceState(outState);
    }

}

File: app/src/main/java/uk/co/jatra/lifecycledialog/MainActivity.kt
package uk.co.jatra.lifecycledialog

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val answerModel = ViewModelProviders.of(this).get(AnswerModel::class.java)

        button.setOnClickListener { view -> LifeCycleDialogFragment().show(supportFragmentManager, "tag") }

        answerModel.dialogAnswer.observe(this, Observer<Answer> { it?.let { it1 -> this.setAnswer(it1) } })
    }

    private fun setAnswer(answer: Answer) {
        textview.text = answer.name
    }
}
